core/arch/arm/kernel/link.mk

```diff
@@ -221,6 +221,26 @@ $(link-out-dir)/tee.bin: $(link-out-dir)/tee-pager.bin \
 		--tee_pageable_bin $(link-out-dir)/tee-pageable.bin \
 		--out $@
 
+all: $(link-out-dir)/tee-header-v2.bin
+cleanfiles += $(link-out-dir)/tee-header-v2.bin
+$(link-out-dir)/tee-header-v2.bin: $(link-out-dir)/tee.bin
```

Comment:
I'd prefer if this wasn't generated from `tee.bin` since that makes it hard to get rid `tee.bin` once we've switched to this new format. Generating from `tee.bin` also has the drawback that there's yet another step where you need to double check when debugging errors related to the generated binary.